objects/src/errors.rs

```diff
@@ -16,22 +16,41 @@ use super::{
     MAX_OUTPUT_NOTES_PER_BATCH, MAX_OUTPUT_NOTES_PER_BLOCK,
 };
 use crate::{
-    accounts::{AccountCode, AccountIdPrefix, AccountStorage, AccountType},
+    accounts::{AccountCode, AccountIdPrefix, AccountStorage, AccountType, StoragePlaceholder},
     block::block_num_from_epoch,
     notes::{NoteAssets, NoteExecutionHint, NoteTag, NoteType, Nullifier},
     ACCOUNT_UPDATE_MAX_SIZE, MAX_INPUTS_PER_NOTE, MAX_INPUT_NOTES_PER_TX, MAX_OUTPUT_NOTES_PER_TX,
 };
 
+// ACCOUNT COMPONENT TEMPLATE ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum AccountComponentTemplateError {
+    #[cfg(feature = "std")]
+    #[error("error trying to deserialize from toml")]
+    DeserializationError(#[source] toml::de::Error),
+    #[error("slot {0} is defined multiple times")]
+    DuplicateSlot(u8),
+    #[error("component storage slots have to start at 0")]
+    StorageSlotsMustStartAtZero,
+    #[error("storage value was not of the expected type {0}")]
+    IncorrectStorageValue(String),
+    #[error("multi-slot entry should contain as many values as storage slots indices")]
+    MultiSlotArityMismatch,
+    #[error("error deserializing component metadata: {0}")]
+    MetadataDeserializationError(String),
+    #[error("component storage slots are not contiguous ({0} is followed by {1})")]
+    NonContiguousSlots(u8, u8),
+    #[error("storage value for placeholder ({0}) was not provided in the map")]
+    PlaceholderValueNotProvided(StoragePlaceholder),
```

Comment:
```suggestion
    #[error("storage value for placeholder {0} was not provided in the map")]
    PlaceholderValueNotProvided(StoragePlaceholder),
```

Nit: Error is a bit nicer without the extra `()`

Reply:
Agreed